mesh/pmesh.cpp

```diff
@@ -3107,6 +3149,8 @@ void ParMesh::Rebalance()
       Nodes = new_nodes;
    }
 
+   have_global_element_offset = false;
```

Comment:
Maybe resetting this could be moved to some clean-up function that already gets called on each mesh change? (This way there is always the danger we'll forget to reset somewhere in future code)

Reply:
Yes, that would be nice.  In searching for places where it was needed, I noticed DeleteFaceNbrData() was a good proxy for where it was needed.  Perhaps both could be put into a function MeshChanged() or something like that.